FBSnapshotTestCase/Categories/UIImage+Compare.m

```diff
@@ -129,4 +115,63 @@ - (BOOL)fb_compareWithImage:(UIImage *)image tolerance:(CGFloat)tolerance
     return imageEqual;
 }
 
+- (BOOL)_comparePixelWithPixelTolerance:(CGFloat)pixelTolerance
+                         referencePixel:(FBComparePixel *)referencePixel
+                             imagePixel:(FBComparePixel *)imagePixel
+{
+    if (referencePixel->raw == imagePixel->raw) {
+        return YES;
+    } else if (pixelTolerance == 0) {
+        return NO;
+    }
+
+    CGFloat redPercentDiff = [self _calculatePercentDifferenceForReferencePixelComponent:referencePixel->pixels.red
+                                                                     imagePixelComponent:imagePixel->pixels.red];
+    CGFloat greenPercentDiff = [self _calculatePercentDifferenceForReferencePixelComponent:referencePixel->pixels.green
+                                                                       imagePixelComponent:imagePixel->pixels.green];
+    CGFloat bluePercentDiff = [self _calculatePercentDifferenceForReferencePixelComponent:referencePixel->pixels.blue
+                                                                      imagePixelComponent:imagePixel->pixels.blue];
+    CGFloat alphaPercentDiff = [self _calculatePercentDifferenceForReferencePixelComponent:referencePixel->pixels.alpha
+                                                                       imagePixelComponent:imagePixel->pixels.alpha];
+
+    return !(redPercentDiff > pixelTolerance ||
+             greenPercentDiff > pixelTolerance ||
+             bluePercentDiff > pixelTolerance ||
+             alphaPercentDiff > pixelTolerance);
+}
+
+- (CGFloat)_calculatePercentDifferenceForReferencePixelComponent:(char)p1
+                                             imagePixelComponent:(char)p2
+{
+    int referencePixelComponent = (unsigned char)p1;
```

Comment:
These should be `NSUInteger` instead of `int`

Reply:
`NSInteger` resolves into `int` or `long` anyway if my understanding is correct, and I thought `int` would have been the safer of the two. Also because it's unlikely we'll need the precision that systems which can offer `long` can give.

Reply:
I'm happy to change it NSUInteger, just let me know what you think.

Reply:
I guess since `ABS` is architecture independent we would want to use it. I originally thought to mention using `NSUInteger` because when I'm writing code my brain wants me to specify the intent of a variable by saying "this will have negative values" or "this will never have negative values"; I usually don't think of it from a precision view point.

What about using `ABS` with just `NSInteger`s instead?

Reply:
We'll do that 👍